compiler/parser.nim

```diff
@@ -91,7 +91,7 @@ proc closeParser(p: var TParser) =
 
 proc parMessage(p: TParser, msg: TMsgKind, arg = "") =
   ## Produce and emit the parser message `arg` to output.
-  lexMessage(p.lex, msg, arg)
+  lexMessageTok(p.lex, msg, p.tok, arg)
```

Comment:
Also I think (if there are not already) maybe it would be worth to have some tests on expected error messages. Just to be sure that similar commit don't cause a regression. I think I will ping you on IRC for some questions about tests, especially these failing compilation.